pkg/proc/stack.go

```diff
@@ -452,14 +488,15 @@ func (it *stackIterator) advanceRegs() (callFrameRegs op.DwarfRegisters, ret uin
 }
 
 func (it *stackIterator) executeFrameRegRule(regnum uint64, rule frame.DWRule, cfa int64) (*op.DwarfRegister, error) {
+	// fmt.Printf("executeFrameRegRule: regnum=%d, rule=%v, cfa=%d\n", regnum, rule, cfa)
 	switch rule.Rule {
 	default:
 		fallthrough
 	case frame.RuleUndefined:
 		return nil, nil
 	case frame.RuleSameVal:
 		if it.regs.Reg(regnum) == nil {
-			return nil, nil
+			return nil, fmt.Errorf("register %d undefined", regnum)
```

Comment:
I'm not sure, reading the specification that we should really return an error here. For example, a frame could have this rule for a register and we could come to it starting from a parked goroutine and then maybe we wouldn't have a value for that register.

Reply:
Yeah, we can just return nil here.